### DISCUSSION TITLE
I can't build a device service with device-sdk-c v3.0.0

### DISCUSSION THREAD

**Discussion by bighb69738** [edgexfoundry/community#207] [Q&A] [answered]
Error message:
In file included from /device-sdk-c/include/devsdk/devsdk.h:17,
                 from template.c:11:
/device-sdk-c/include/devsdk/devsdk-base.h:17:10: fatal error: iot/data.h: No such file or directory
   17 | #include "iot/data.h"
      |          ^~~~~~~~~~~~
    
    
 ### Alpine

```
wget https://iotech.jfrog.io/artifactory/api/security/keypair/public/repositories/alpine-release -O /etc/apk/keys/alpine.dev.rsa.pub
echo "https://iotech.jfrog.io/artifactory/alpine-release/v3.16/main" >> /etc/apk/repositories
apk update
apk add iotech-iot-1.5-dev
```

The provided instructions have already been added to my Dockerfile. Do I need to add anything else to find the missing files?

**Comment by cloudxxx8** [comment]
you can check if there is anything different from the other C Device Service build script
https://github.com/edgexfoundry/device-bacnet-c/tree/main/scripts

**Reply by bighb69738** [reply]
Hi @cloudxxx8 ,

It is my docker file

ARG BASE=edgex-csdk-base:2.1.0
FROM ${BASE} as builder
RUN apk add --update --no-cache build-base git gcc cmake make linux-headers yaml-dev libmicrohttpd-dev curl-dev util-linux-dev ncurses-dev libmodbus automake autoconf libtool

˙˙˙
FROM ${BASE}
WORKDIR /
COPY --from=builder /tmp/sdk .
COPY scripts /device-sdk-c/scripts
COPY src /device-sdk-c/src/
COPY VERSION /device-sdk-c/VERSION
COPY include /device-sdk-c/include
COPY src/c/examples/template.c /device-sdk-c/
RUN export CSDK_DIR=/build/release/_CPack_Packages/Linux/TGZ/csdk-3.0.0
RUN mkdir -p /device-sdk-c/build

WORKDIR /device-sdk-c
RUN gcc -I/build/release/_CPack_Packages/Linux/TGZ/csdk-3.0.0/include -L/build/release/_CPack_Packages/Linux/TGZ/csdk-3.0.0/lib -o device-example-c template.c -lcsdk -lpthread -lmodbus -ljson-c -lcurl

and the error message is 
In file included from /build/release/_CPack_Packages/Linux/TGZ/csdk-3.0.0/include/devsdk/devsdk.h:17,
                 from template.c:11:
/build/release/_CPack_Packages/Linux/TGZ/csdk-3.0.0/include/devsdk/devsdk-base.h:17:10: fatal error: iot/data.h: No such file or directory
   17 | #include "iot/data.h"
      |          ^~~~~~~~~~~~
so where is the file "iot/data.h" ?

**Reply by cloudxxx8** [reply, marked as answer]
Please refert to the build script.
There are more than one docker file, and some more dependencies are required for the build.
https://github.com/edgexfoundry/device-bacnet-c/blob/ca615f754fc2c689e0c4e16d61f784c0b2379ced/scripts/Dockerfile.alpine-base#L22
